docs/architecture-decisions/adr002-default-catalog-file-format.md

```diff
@@ -0,0 +1,245 @@
+  <tbody>
+    <tr>
+      <td valign="top"><code>apiVersion</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The version of specification used for this kind. This field is used for
+          being able to evolve the format, and the tuple of <code>apiVersion</code>
+          and <code>kind</code> are enough for a parser to know how to interpret
+          the contents of the document.
+        </p>
+        <p>
+          Backstage specific entities have an <code>apiVersion</code> that is
+          prefixed with <code>backstage.io/</code>.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>kind</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The high level entity type. In the first version of the software catalog,
+          we will focus on the <code>Component</code> kind.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>metadata</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          Metadata about the entity, such as its name, that are not directly part of
+          the entity specification itself. See below for more details about this structure.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>spec</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The actual specification data that describes the entity.
+        </p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+The `metadata` root field has the following nested structure.
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Required</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td valign="top"><code>name</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The name of the entity. Names must be unique per kind, within a given namespace.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>namespace</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The `name` of a namespace that the entity belongs to. This field is optional.
+        </p>
+        <p>
+          Namespaces may also be part of the catalog, and are <code>v1</code>/<code>Namespace</code>
+          objects, i.e. not Backstage specific but the same as in Kubernetes.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>description</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          A human readable description of the entity, to be shown in Backstage. May contain
+          markdown.
```

Comment:
I don't think allowing markdown here is gonna end up well, too many renderers that won't be bothered supporting that :grin:. Maybe a few things like **bold**, that \*\*kinda\*\* reads alright anyway?

Reply:
I would actually like to leave out the markdown bit. We have it internally and it has been asked for and used, but it also muddles things very quickly if overused. And then you start using github-flavored markdown and you get even more locked in etc. Most asks for markdown were for cases that probably were better suited for separate documentation text, dedicated metadata fields, and similar, anyway.

Reply:
Maybe we could even limit it to a tagline or somesuch, similar to what Github does. We can always pull in a short or long-form description from the README or techdocs as additional plugins, not sure if it's worth storing it in the catalog.

Reply:
Left as a free form string for now, but we can add validation to the string (length limit, no newlines or whatever) later